core/server/master/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -470,6 +472,17 @@ void setAttribute(AlluxioURI path, SetAttributeOptions options)
    */
   void scheduleAsyncPersistence(AlluxioURI path) throws AlluxioException, UnavailableException;
 
+  /**
+   * Update the operation mode for the given ufs path under one or more mount points.
+   *
+   * @param ufsPath the ufs path
+   * @param ufsMode the ufs operation mode
+   * @throws InvalidPathException if ufs path is not used by any mount point
+   * @throws InvalidArgumentException if arguments for the method are invalid
+   */
+  void updateUfsMode(String ufsPath, UnderFileSystem.UfsMode ufsMode) throws InvalidPathException,
```

Comment:
Can `ufsPath` be `AlluxioURI` like the others in the interface? And maybe rename to `ufsRoot` or something.